CHANGELOG.md

```diff
@@ -1,5 +1,113 @@
+
+```json
+{
+    "mainWindowUrl": "./index.html"
+}
+```
+
+Place this file into the application root directory.
+
+Now you are ready to run tests. Specify the `electron` browser name and the application path
+when launching tests from the command line or API.
+
+```sh
+testcafe "electron:/home/user/electron-app" "path/to/test/file.js"
+```
+
+```js
+testCafe
+    .createRunner()
+    .src('path/to/test/file.js')
+    .browsers('electron:/home/user/electron-app')
+    .run();
+```
+
+To learn more about the Electron browser provider, see the plugin [readme](https://github.com/DevExpress/testcafe-browser-provider-electron).
+
+#### :gear: Concurrent test execution ([#1165](https://github.com/DevExpress/testcafe/issues/1165))
+
+To save your time spent on testing, we have added the capability to run tests concurrently in a single browser.
+
+By default, when you run tests in one or several browsers,
+TestCafe opens a single instance of each browser and runs all test sequentially within each instance.
+
+*Concurrency* allows TestCafe to open multiple instances of the same browser
+and distribute the test batch among these instances so that tests run in parallel.
+
+You can enable concurrency with the `-c` command line option or the `runner.concurrency()` API method.
+When doing so, specify the number of instances you wish to invoke.
+
+```js
+testcafe -c 3 chrome tests/test.js
+```
+
+```js
+var testRunPromise = runner
+    .src('tests/test.js')
+    .browsers('chrome')
+    .concurrency(3)
+    .run();
+```
+
+For details, see [Concurrent Test Execution](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/concurrent-test-execution.html).
+
+#### :gear: Further improvements in automatic waiting mechanism ([#1521](https://github.com/DevExpress/testcafe/issues/1521))
+
+We have enhanced the waiting mechanism behavior in certain scenarios where you still used to need `wait` actions.
+Now automatic waiting is much smarter and chances that you need to wait manually are diminished.
+
+#### :gear: User roles preserve the local storage ([#1454](https://github.com/DevExpress/testcafe/issues/1454))
+
+TestCafe now saves the local storage state when switching between roles, so that when you switch back you get the same local storage content you left.
+
+This is useful for testing websites that perform authentication via local storage instead of cookies.
+
+#### Internal Improvements
+
+* `StorageSandbox` is shared to the client API with `clear` and `lock` methods added. ([testcafe-hammerhead/#1189](https://github.com/DevExpress/testcafe-hammerhead/issues/1189))
+* The number of failed cookie-sync messages is now limited. ([testcafe-hammerhead/#1193](https://github.com/DevExpress/testcafe-hammerhead/issues/1193))
+* `Hammerhead.backupStorages` and `Hammerhead.restoreStorages` methods are added to the client API([testcafe-hammerhead/#1125](https://github.com/DevExpress/testcafe-hammerhead/issues/1125))
```

Comment:
Do we give links to these methods? 
Explaining them all can be too much, but at least links to see what they're for.